scipy/sparse/linalg/propack.py

```diff
@@ -0,0 +1,336 @@
+        respectively), `blocksize` controls how large a fraction of the
+        work is done via fast BLAS level 3 operations.  A larger blocksize
+        may lead to faster computation at the expense of greater memory
+        consumption.  `blocksize` must be ``>= 1``.  Default is `16`.
+    min_relgap : float, optional
+        The smallest relative gap allowed between any shift in IRL mode.
+        Default is `0.001`.  Accessed only if ``irl_mode=True``.
+    shifts : int, optional
+        Number of shifts per restart in IRL mode.  Default is determined
+        to satisfy ``k <= min(kmax-shifts, m, n)``.  Must be
+        >= 0, but choosing 0 might lead to performance degredation.
+        Accessed only if ``irl_mode=True``.
+    maxiter : int, optional
+        Maximum number of restarts in IRL mode.  Default is `1000`.
+        Accessed only if ``irl_mode=True``.
+
+    Returns
+    -------
+    u : ndarray
+        The top `k` left singular vectors, ``shape == (A.shape[0], 3)``,
+        returned only if ``compute_u=True``.
+    sigma : ndarray
+        The top `k` singular values, ``shape == (k,)``
+    vt : ndarray
+        The top `k` right singular vectors, ``shape == (3, A.shape[1])``,
+        returned only if ``compute_v=True``.
+    sigma_bound : ndarray
+        the error bounds on the singular values sigma, returned only if
+        ``full_output=True``.
+
+    Examples
+    --------
+    >>> from scipy.sparse.linalg import svdp
+    >>> A = np.random.random((10, 20))
+    >>> u, sigma, vt = svdp(A, 3)
+    >>> np.set_printoptions(precision=3, suppress=True)
+    >>> print(abs(np.dot(u.T, u)))
+    [[ 1.  0.  0.]
+     [ 0.  1.  0.]
+     [ 0.  0.  1.]]
+    >>> print(abs(np.dot(vt, vt.T)))
+    [[ 1.  0.  0.]
+     [ 0.  1.  0.]
+     [ 0.  0.  1.]]
+    """
+
+    which = which.upper()
+    if which not in {'LM', 'SM'}:
+        raise ValueError("`which` must be either 'LM' or 'SM'")
+    if not irl_mode and which == 'SM':
+        raise ValueError("`which`='SM' requires irl_mode=True")
+
+    aprod = _AProd(A)
+    typ = aprod.dtype.char
+
+    try:
+        lansvd_irl = _lansvd_irl_dict[typ]
+        lansvd = _lansvd_dict[typ]
+    except KeyError:
+        # work with non-supported types using native system precision
+        if np.iscomplexobj(np.empty(0, dtype=typ)):
+            typ = np.dtype(complex).char
+        else:
+            typ = np.dtype(float).char
+        lansvd_irl = _lansvd_irl_dict[typ]
+        lansvd = _lansvd_dict[typ]
+
+    m, n = aprod.shape
+    if (k < 1) or (k > min(m, n)):
+        raise ValueError("k must be positive and not greater than m or n")
+
+    if kmax is None:
+        kmax = 5*k
+    if maxiter is None:
+        maxiter = 1000
+
+    # guard against unnecessarily large kmax
+    kmax = min(m + 1, n + 1, kmax)
+    if kmax < k:
+        raise ValueError(
+            f"kmax must be greater than or equal to k: {kmax} < {k}")
```

Comment:
```suggestion
            f"kmax must be greater than or equal to k, but kmax ({kmax}) < k ({k})")
```
The colon suggests that the following statement will expand on the previous one, but it may not be immediately clear to the user what these numbers mean.